### ISSUE TITLE
Value-returning scopes typed as Builder<Model>, ignoring the scope's declared return (callScope ?? $this)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1053]
**Describe the bug**

A local scope is not required to return the builder. Laravel's `Builder::callScope()` returns `$scope(...) ?? $this`, so a scope whose body `return`s a value (e.g. `->first()`, a count, a collection) propagates that value to the caller. The plugin types every forwarded scope call as `Builder<Model>`, ignoring the scope's declared return type. Downstream this produces false positives on the real return: a null check on a `?Model`-returning scope is flagged `TypeDoesNotContainNull`/redundant, and member access goes through the wrong type.

**Code to reproduce**
```php
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;

class Post extends Model
{
    /** @param Builder<self> $query */
    public function scopeFirstPublished(Builder $query): ?self
    {
        return $query->where('published', true)->first();
    }
}

function probe(): void
{
    $r = Post::query()->firstPublished();
    /** @psalm-check-type-exact $r = Post|null */   // actual: Builder<Post>

    if ($r === null) { return; }                    // actual: TypeDoesNotContainNull (false positive)
}
```

**Impacted Versions**
```text
psalm/plugin-laravel  dev-master @ 939d9b6f
vimeo/psalm           7.0.0-beta19
laravel/framework     v13.15.0
PHP 8.4
```

**Psalm config**
```xml
<?xml version="1.0"?>
<psalm errorLevel="3" findUnusedCode="false" xmlns="https://getpsalm.org/schema/config">
    <projectFiles><directory name="app"/></projectFiles>
    <plugins><pluginClass class="Psalm\LaravelPlugin\Plugin"/></plugins>
</psalm>
```

**Additional context**

`Builder<Model>` is correct for the common void/builder-returning scope (the `?? $this` fallback). The bug is only for scopes whose declared return is non-void and not a builder. Proposed fix: when the scope's declared return type is non-void and not a `Builder` subtype, return `declared | Builder<Model>` (dropping `null` from the declared part, since a null scope result falls back to the builder). The same `?? $this` semantics affect `Builder::scopes(...)` (typed `static|mixed` upstream); a consistent decision should cover both. Larastan also fabricates `Builder` here, so this would be a precision improvement over both tools.
